src/Common/SipHash.h

```diff
@@ -29,6 +31,61 @@ namespace ErrorCodes
 {
     extern const int LOGICAL_ERROR;
 }
+
+template <typename T>
```

Comment:
I have the impression that your `TransformEndianness` header ([here](https://github.com/ClickHouse/ClickHouse/pull/50405/files#diff-acbad3e9ab9b827fecd7e6759b3ce7cbf3e5c759699a8b6be292e8643d6e7e0c)) could be re-used in this PR.